### PR TITLE
fix field selector generation

### PR DESCRIPTION
**Pull request by smonero**
### Description

due to https://github.com/kubernetes/kubernetes/issues/32946 we need to do the field selection multiple times when having multiple types of items to get

### Testing Performed

unit

### GitHub Issue

Fixes #

### CODE REVIEW

File: backend/service/k8s/event.go [hunk 1]
```diff
@@ -91,15 +88,21 @@ func (s *svc) ListNamespaceEvents(ctx context.Context, clientset, cluster, names
 	}
 
 	// Note if field selector is blank it will return everything
-	eventList, err := cs.CoreV1().Events(cs.Namespace()).List(ctx, metav1.ListOptions{FieldSelector: convertTypesToFieldSelector(types)})
-	if err != nil {
-		return nil, err
+	totalEventList := []corev1.Event{}
+	fs := convertTypesToFieldSelectors(types)
+	for _, f := range fs {
```
Comment (scarlettperry) [inline backend/service/k8s/event.go:93]: if the request didn't pass filter types then this wouldn't make make the api call as `fs` would be empty, unless im misunderstanding

File: backend/service/k8s/event.go [hunk 2]
```diff
@@ -91,15 +88,21 @@ func (s *svc) ListNamespaceEvents(ctx context.Context, clientset, cluster, names
...
+		eventList, err := cs.CoreV1().Events(cs.Namespace()).List(ctx, metav1.ListOptions{FieldSelector: f})
```
Comment (scarlettperry) [inline backend/service/k8s/event.go:94]: since we're now collecting the event results of separate API calls, do we need to sort the events list by timestamp?

File: backend/service/k8s/event.go [hunk 3]
```diff
@@ -34,51 +35,65 @@ func (s *svc) ListEvents(ctx context.Context, clientset, cluster, namespace, obj
 	return events, nil
 }
 
-func ProtoForEvent(cluster string, k8sEvent *corev1.Event) *k8sapiv1.Event {
-	clusterName := GetKubeClusterName(k8sEvent)
-	if clusterName == "" {
-		clusterName = cluster
```
Comment (scarlettperry) [inline backend/service/k8s/event.go:40]: curious, why was this removed? (https://github.com/lyft/clutch/pull/2704/files)

**Review by scarlettperry — APPROVED** [review]

File: backend/service/k8s/event.go (hunk 3 above)
Comment (smonero) [reply backend/service/k8s/event.go]: oops yeah I accidently removed that when i merged in main branch
